### PR TITLE
RMF: Create compressed datasets

### PR DESCRIPTION
**Pull request by drons**
* Add LZW encoder
 * Add DEM encoder
 * Add JPEG encoder
 * Add strict decompressed data size check
 * Fix LZW decoder test. Actually, the old files were not compressed. The coverage of rmflzw.cpp was empty: https://rawgit.com/rouault/gdalautotest-coverage-results/master/coverage_html/frmts/rmf/index.html

### CODE REVIEW

File: gdal/frmts/rmf/rmfjpeg.cpp [hunk 1]
```diff
@@ -123,4 +124,93 @@ int RMFDataset::JPEGDecompress(const GByte* pabyIn, GUInt32 nSizeIn,
...
+    char* apszJpegOptions[2] = {szQuality, nullptr};
+
+    GDALDatasetH   hJpeg = GDALCreateCopy(hJpegDriver, osTmpFilename, poMemDS,
+                                          0, apszJpegOptions, nullptr, nullptr);
+    GDALClose(poMemDS);
+
+    if(hJpeg == nullptr)
+    {
+        CPLError(CE_Failure, CPLE_AppDefined,
+                 "RMF JPEG: Error compress JPEG tile");
```
Comment (rouault) [inline gdal/frmts/rmf/rmfjpeg.cpp]: VSIUnlink(osTmpFilename) missing here

File: gdal/frmts/rmf/rmflzw.cpp [hunk 2]
```diff
@@ -232,9 +288,104 @@ int RMFDataset::LZWDecompress( const GByte* pabyIn, GUInt32 nSizeIn,
...
+{
+    if( pabyIn == nullptr ||
+        pabyOut == nullptr ||
+        nSizeIn == 0 )
+        return 0;
+
+    // Allocate space for the new table and pre-fill it
+    LZWStringTab *poCodeTab =
+        (LZWStringTab *)CPLMalloc(TABSIZE * sizeof(LZWStringTab));
+    if( !poCodeTab )
```
Comment (rouault) [inline gdal/frmts/rmf/rmflzw.cpp]: this test is useless. In case of failure CPLMalloc() aborts

File: gdal/frmts/rmf/rmflzw.cpp [hunk 3]
```diff
@@ -109,23 +111,77 @@ static void LZWUpdateTab(LZWStringTab *poCodeTab, GUInt32 iPred, char bFoll)
...
     if( pabyIn == nullptr ||
         pabyOut == nullptr ||
-        nSizeOut < nSizeIn ||
         nSizeIn < 2 )
         return 0;
-
+    GByte* const pabyOutBegin = pabyOut;
     // Allocate space for the new table and pre-fill it
     LZWStringTab *poCodeTab =
         (LZWStringTab *)CPLMalloc( TABSIZE * sizeof(LZWStringTab) );
```
Comment (rouault) [inline gdal/frmts/rmf/rmflzw.cpp]: the below test is useless

Comment (drons) [reply gdal/frmts/rmf/rmflzw.cpp]: @rouault  What is better to use next time CPLMalloc or VSIMalloc? Any suggestions?

Comment (rouault) [reply gdal/frmts/rmf/rmflzw.cpp]: CPLMalloc() is fine for small allocations (let's say ~ 1 KB) that should normally not fail

**Pull request merged**